packages/web-twig/src/Resources/components/Link/Link.twig

```diff
@@ -4,17 +4,30 @@
 {%- set _href = props.href -%}
 {%- set _isDisabled = props.isDisabled | default(false) -%}
 {%- set _isUnderlined = props.isUnderlined | default(false) -%}
+{%- set _underline = props.underline | default('hover') -%}
+
+{# Variables #}
+{%- set rootUnderlineClassName = '' %}
 
 {# Class names #}
 {%- set _colorClassName = _spiritClassPrefix ~ 'link-' ~ _color -%}
 {%- set _rootDisabledClassName = _isDisabled ? _spiritClassPrefix ~ 'link-disabled' : null -%}
-{%- set _rootUnderlinedClassName = _isUnderlined ? _spiritClassPrefix ~ 'link-underlined' : null -%}
+{%- if _isUnderlined or _underline is same as('always') %}
+    {%- set rootUnderlineClassName = _spiritClassPrefix ~ 'link-underlined' -%}
+{% elseif _underline is same as('never') %}
+    {%- set rootUnderlineClassName = _spiritClassPrefix ~ 'link-not-underlined' -%}
+{% endif %}
+
 
 {# Miscellaneous #}
 {%- set _styleProps = useStyleProps(props) -%}
-{%- set _classNames = [ _colorClassName, _rootDisabledClassName, _rootUnderlinedClassName, _styleProps.className ] -%}
+{%- set _classNames = [ _colorClassName, _rootDisabledClassName, rootUnderlineClassName, _styleProps.className ] -%}
 {%- set _allowedAttributes = [ 'target', 'title' ] -%}
 
+{% if _isUnderlined is defined %}
+    {% deprecated '"isUnderlined" property will be replaced in the next major version. Please use "underline" instead.' %}
```

Comment:
```suggestion
    {% deprecated '"isUnderlined" property will be removed in the next major version. Please use "underline" instead.' %}
```

Reply:
I would leave it like this because it is a generated message from `useDeprecationMessage`. It would be weird if we had two different messages.